Binary packages

gartung opened the pull request:
Second try

trws committed:
go and a basic go package

trws committed:
track whether a package was installed "explicitly"

Adds a new attribute in the database to track whether a package was
installed explicitly or not, where explicitly is the user running `spack
install <package>` and implicitly is it being installed as a dependency.
It also adds arguments to `spack find` to find these packages such that
it should be possible to query the packages that were installed
implicitly and are not currently depended upon any longer.

xjrc committed:
Updated and added new versions to the METIS/ParMETIS packages.
- Added a 'url_for_version' function to the METIS/ParMETIS packages.
- Added installation support for METIS@5.0.2 and ParMETIS@4.0.2.
- Changed the 'double' variant to 'real64' to make it more consistent
  with the 'idx64' variant naming.
- Removed an unnecessary dependency on gdb.

xjrc committed:
Improved MPI path specification for ParMETIS using PR #893 features.

trws committed:
moving go packages to the new path

becker33 committed:
fixed pkgconfig issue

mamelara committed:
Changed _proper_compiler_spec to return filtered list. Apparently filter returns a list rather than filtering the original list. Lead to compiler finding problems if user specified frontend compilers

brettviren committed:
Move from str.format() to string.Template.

brettviren committed:
Remove problematic tests, deal with this issue outside of PR #869.

davydden committed:
sed: add package

brettviren committed:
Pick a better name for the 'spec' parameters.

brettviren committed:
Document  command.

Commit:
netcdf-cxx4: Update to latest released version 4.3.0

trws committed:
complete lua rework, also module path fix

This is a complete rework of the lua package, it also allows the
environment modification classes to handle paths that are not separated
by colons, and uses the support for same in TCL modules as well.

The biggest difference is the handling for lua extension packages, which
now have their paths set correctly by the lua parent package, and have
access to both lua and luarocks as installation tools. See the luaposix
package for what should be required for most lua packages after this.

trws committed:
fix merge error

Commit:
Updated turbomole package file.

This PR updates the turbomole package file but does not introduce any
new funtionality. The updtes are:
- use spack interface for subprocess.PIPE and subprocess.Popen
- clean up based on flake8
- remove some extra whitespace

trws committed:
changes to appease flake8 (mostly) for lua package

trws committed:
changes to appease flake8 for lua-luaposix package

trws committed:
appeasing flake8 in environment

trws committed:
appeasing the great vengeful flake8 gods

trws committed:
yet more sacrifices to the god of short-lines

trws committed:
adding go_compiler providers

trws committed:
go rework

This commit includes:
* a new go package that uses gccgo to bootstrap the go toolchain
* env support added to Executable
* a new Go fetch strategy that uses `go get` to fetch a package and all
  of its deps
* A platinum searcher package that leverages the new go package and
  fetch strategy

trws committed:
removing comments

also randomly changed unnecessary licence block

davydden committed:
ghostscript: url was changed upstream

davydden committed:
graphviz: add missing dependency and fix installation on darwin

davydden committed:
doxygen: add graphviz variant; enable flex and bison on OSX and they build fine

trws committed:
environment: fixed comment and default separator

trws committed:
remove debug print

trws committed:
go cleanup, adding hub

davydden committed:
indentation fixes

tgamblin committed:
Merge pull request #957 from glennpj/turbomole_subprocess

Updated turbomole package file.

tgamblin committed:
Merge pull request #959 from davydden/bugfix/doxygen_ghostscript_graphviz

Bugfix/doxygen ghostscript graphviz

trws committed:
fix linking issue on ubuntu

Evidently some readline variants are built with only a dynamic dependency on
ncurses, this addresses that problem for such systems.

trws committed:
formatting to appease flake8

trws committed:
fixing more flake8 errors that only appear on travis...

brettviren committed:
flake8-clean.

brettviren committed:
Merge remote-tracking branch 'upstream/develop' into view-fix-merge

trws committed:
handle explicit when requesting to install already installed packages

trws committed:
initial rust support

trws committed:
add GOPATH support

Also fixes flex to depend on bison for its yacc implementation.

becker33 committed:
Added more compilers to mock packages tests

Commit:
Placate flake8 demands.

becker33 committed:
Fixed config tests for new compiler config format

tgamblin committed:
Merge pull request #953 from citibeth/150514-NetCDFCXX4

netcdf-cxx4: Update to latest released version 4.3.0

xjrc committed:
Updated style checking to ignore refinition errors for '@when' fxns.
Updated the Scotch package to conform to PEP8 standards and to test style checking improvements.

xjrc committed:
Added line limit exemptions for 'variant' and 'version' directives.
Updated the 'scotch' package to use oneline 'variant' directives.

trws committed:
rework for gopath and bootstrapping

trws committed:
flake8-related cleanup

tgamblin committed:
Merge pull request #958 from trws/golang-new

Golang new

verolero86 committed:
add a flag to openmpi to allow users to disable vt support

becker33 committed:
Merge pull request #952 from davydden/package/sed

sed: add package

verolero86 committed:
the right flag here is +, the default behavior is controlled in the variant

eschnett committed:
Remove duplicate providers that can concretize to the same result

Patch generously provided by @becker33.

Closes #928.

tgamblin committed:
Merge pull request #963 from eschnett/eschnett/duplicate-providers

Remove duplicate providers that can concretize to the same result

tgamblin committed:
Merge pull request #955 from trws/lua-rework

complete lua rework

davydden committed:
tbb: add 4.4.4; raise InstallError when building 4.4.3 with GCC6.1

tgamblin committed:
Merge pull request #868 from trws/explicitly-installed

track whether a package was installed "explicitly"

davydden committed:
formatting

gartung committed:
use only otool and install_name_tool

tgamblin committed:
Tag Spack version 0.9

- Tagging a version to include all that's happened since 0.8.17

- This is in anticipation of merging `features/cflags` and
  `features/newarch`, which will break the spec format.

- 1.0 will likely come after those.

gartung committed:
fixes

tgamblin committed:
Merge branch 'features/cflags' into develop

Conflicts:
	lib/spack/spack/cmd/find.py
	var/spack/repos/builtin/packages/ghostscript/package.py

tgamblin committed:
Merge pull request #961 from xjrc/features/flake8-improvements

Enhancement Proposal: Exempt '@when' Functions from Style Redefinition Errors

hegner commented on lib/spack/spack/relocate.py:
```diff
@@ -44,6 +46,64 @@ def get_existing_rpath(path_name, patchelf_executable):
 
+def change_dylib(path_name, rpaths):
+    """
+    Return the RPATHS in given file as a list of strings.
```
Wrong comment ;-)

hegner commented on lib/spack/spack/relocate.py:
```diff
@@ -44,6 +46,64 @@ def get_existing_rpath(path_name, patchelf_executable):
+    last_cmd = None
+    path = ''
+    deps = []
+    for line in output.split('\n'):
```
Could you give an example of what the regular expressions are looking for? And wouldn't much simpler code that checks whether things start with the orig_path? You don't need complex code for that.

hegner commented on lib/spack/spack/relocate.py:
```diff
@@ -117,6 +177,8 @@ def relocate_binary(path_name, old_dir, new_dir, patchelf_executable):
     new_rpath  = substitute_rpath(orig_rpath, old_dir, new_dir)
     modify_rpath(path_name, orig_rpath, new_rpath, patchelf_executable)
+    if platform.system() == 'Darwin':
+       change_dylib(path_name, orig_rpath)
```
method name is misleading. Because you don't change a dylib in complete, but just special parts of it.

hegner commented on lib/spack/spack/relocate.py:
```diff
@@ -117,6 +177,8 @@ def relocate_binary(path_name, old_dir, new_dir, patchelf_executable):
     orig_rpath = get_existing_rpath(path_name, patchelf_executable)
     new_rpath  = substitute_rpath(orig_rpath, old_dir, new_dir)
     modify_rpath(path_name, orig_rpath, new_rpath, patchelf_executable)
```
maybe we can already pull the darwin specific parts out if modify_rpath and put all together in one function for linux and one for mac?

alalazo committed:
fixes #967

alalazo committed:
fixes #967, flake8 approved

Rombur committed:
Add boost 1.61.0

tgamblin committed:
Merge pull request #968 from epfl-scitas/find_always_zero_pkgs

Fix : spack find

tgamblin committed:
Merge pull request #948 from davydden/bugfix/astyle

astyle: fix filter command which failed on OSX with python 2.7.10

tgamblin committed:
Merge pull request #969 from Rombur/boost

Add boost 1.61.0

tgamblin committed:
Merge pull request #966 from verolero86/develop

Add option to disable VT support in openmpi package

eschnett committed:
Auto-detect verbs location

trws committed:
updates for rust-bindgen

adamjstewart committed:
Add STREAM benchmark package

trws committed:
initial docs for find and flake8 cleanup

trws committed:
peren-nesting fix in spec

becker33 committed:
Partial merge of mainline develop

tgamblin committed:
Merge pull request #977 from trws/explicit-docs

initial docs for find and flake8 cleanup

gartung committed:
using @rpath was a bust; switched replacing olddir with newdir; add option to Darwin ld to pad rpath entries as much as possible for long relocation paths:

gartung committed:
revert change to cc script

becker33 committed:
Finished merge on these files

mamelara committed:
got rid of ifdef

mamelara committed:
Got rid of ifdefs, changed parameters for Compiler: added kwargs

mamelara committed:
Moved searching of flags into get compilers in compilers_for_spec

becker33 committed:
more merge work

becker33 committed:
more merge work

becker33 committed:
more merge work

mamelara committed:
Changed arch tests to new spec target= os=

mamelara committed:
Changed unsatisfiable architecture test

mamelara committed:
Merge branch 'features/newarch' of https://github.com/NERSC/spack into features/newarch

becker33 committed:
Fixed the first set of merge bugs

becker33 committed:
Fixed some of the bugs

hegner commented:
@gartung - thanks. Can you remove the ROOT specific part from this PR? I really want to have only the new feature. Thanks!

gartung commented:
Removed the change to freetype because of root

brettviren committed:
Remove `view print` for now.

nrichart committed:
Adding +metis variant to scotch + minor changes

xjrc committed:
Updating 'metis' and 'parmetis' to conform to PEP8 standards.

becker33 committed:
fixed spec syntax test

gartung committed:
fix bug that kept LC_LOAD_DYLIB from being changed

trws committed:
fixing warning for package fetch

mamelara committed:
Adding link_paths to craype compilers

nrichart committed:
changes to simplify the esmumps variant for version < 6.0.0 + typos corrections, according to the remarks on PR #979

nrichart committed:
changes scotch package to make he coverage test pass

nrichart committed:
missing @ in version definition

mamelara committed:
Changed rpaths method so that it only grabs rpaths using the compiler module name instead of PrgEnv-x module name

tgamblin committed:
Merge pull request #979 from epfl-scitas/packages/scotch+metis

Adding `+metis` variant to scotch + changes for esmumps

tgamblin committed:
Merge pull request #964 from davydden/update/tbb

tbb: add 4.4.4; raise InstallError when building 4.4.3 with GCC6.1

alalazo committed:
find : added unit tests, exits with 1 if -E and -e are given

alalazo committed:
autopep8 : fixed style

mamelara committed:
compiler list is sorted in descending order, in order to get most up to date version we check from the beginning of the list

tgamblin committed:
update documentation.

trws committed:
rust compiler/builder package, new llvm variants

tgamblin committed:
Fixes #980: really slow spack spec performance

- `spack spec dealii` now takse seconds rather than 18 minutes.

- Slow hash (`dag_hash`) was accidentally inserted into the fast hash
  (`__hash__`).

- Do not put the slow hash in the fast hash.

davydden committed:
octave: add missing dependencies and update to 4.0.2

davydden committed:
octave: fix formatting

gartung commented:
Submit directly to LLNL/spack

gartung committed:
fix incorrect change

gartung committed:
typo

mamelara committed:
Started writing documentation for architecture. Might need a couple more drafts

becker33 committed:
Fixed errors caught by spec and concretize tests

becker33 committed:
'Ugly merge because I mishandled git
Merge branch 'save' into features/newarch

becker33 committed:
Fixed a couple more bugs

becker33 committed:
fixed bug where earlier test was blowing away environment, causing an error on which('modulecmd')

tgamblin committed:
Merge pull request #985 from davydden/update/octave

octave: add missing dependencies and update to 4.0.2

lee218llnl committed:
updated launchmon url and version

xjrc committed:
Fixed a few linking/testing bugs in the METIS@:4 install method.

xjrc committed:
Added staging support and improved error handling for METIS@:4 installs.

xjrc committed:
Added preliminary improved testing code for METIS@:4.

becker33 committed:
updated documentation for cflags PR

becker33 committed:
(hopefully) final fixes of the merge

xjrc committed:
Fixing a small bug with metis@4+shared installs on the Mac platform.

becker33 committed:
added test platform

tgamblin committed:
Merge pull request #982 from trws/rust

Rust

becker33 committed:
Fixed test config to work on yosemite

becker33 committed:
fixed mock config to use truncated versions of linux distro versions

davydden committed:
dealii: don't hardcode blas and mpi; minor fixes to dependencies

mamelara committed:
Changed comparison operator so that we don't get caught in an infinite loop

becker33 committed:
made yaml format backwards compatible

becker33 committed:
fixed compatibility work

davydden committed:
dealii: formatting

becker33 committed:
Allow 'arch=' syntax for specs to allow copying from output

becker33 committed:
fixed compiler config init logic

mamelara committed:
Corrected parantheses wrapping

xjrc committed:
Added a style checking exemption for lambda expressions.

tgamblin committed:
Merge pull request #993 from xjrc/packages/metis

Update Package : METIS/ParMETIS

tgamblin committed:
Merge pull request #992 from davydden/update/dealii

Update/dealii

becker33 committed:
Fixed backwards compatibility for compilers.yaml config file

tgamblin committed:
Merge pull request #975 from adamjstewart/features/stream

Add STREAM benchmark package

tgamblin committed:
Merge pull request #989 from LLNL/features/cflags-documentation

updated documentation for cflags PR

tgamblin committed:
Merge pull request #981 from epfl-scitas/tests/unit_test_for_968

find : add unit tests

tgamblin committed:
Merge pull request #973 from eschnett/eschnett/openmpi-verbs

Auto-detect verbs location

becker33 committed:
Fixed the external modules bug

becker33 committed:
Fixed typo in linux platform

lee218llnl committed:
updated libdwarf version and dwarfdump build

lee218llnl committed:
Merge branch 'develop' of https://github.com/llnl/spack into develop

becker33 committed:
re-merged mainline develop

mamelara committed:
Added external module check in normalize to avoid infinite loop

mamelara committed:
Adding test for external modules

mamelara committed:
Adding external to mock config

mamelara committed:
Adding mock package file for external module

mamelara committed:
Added elcap compilers

mamelara committed:
Added conditional skipIf to test_external_modules, darwin machines don't have tcl modules so skip this test and let user know why

tgamblin committed:
Performance boost: reduce instantiations of re.Scanner

- Lexer is the same for every Spec parser in spack, so don't build it
  every time.

- This improves time to import package.py files a lot, as a Lexer
  doesn't have to be constructed for every spc in the packages.

- To concretize dealii:
  - Before: ~20 sec
  - After:  ~6 sec

adorostkar committed:
additional versions for Dealii

Included additional versions for Deal.II package.

tgamblin committed:
Merge pull request #999 from ashkan2200/develop

Dealii, Versions added

adorostkar committed:
Merge remote-tracking branch 'LLNL/develop' into develop

adorostkar committed:
dealii, test for specific version

Specified that test case “step-40” is only to be run when using dealii
version 8.4.0 or later

adorostkar committed:
Step-40 test for 8.3.0 and earlier

Integrated part of step-40 to be tested for earlier versions of dealii

adorostkar committed:
dealii, Added comment

Added comment for why step-40 is done differently for dealii 8.4.0 and
later

mamelara committed:
Added a conditional to check if package is external

mamelara committed:
Changed platform to py_platform to avoid name collisions. The naming collisions resulted in an error on linux machines

mamelara committed:
Added extra check on linux machines, since most linux machines don't have module system

mamelara committed:
Quick fix for mock compilers.yaml to work on linux machines.

tgamblin committed:
Merge branch 'develop' into features/newarch

tgamblin committed:
More flexible reading of specs from DB, formatting.

adorostkar committed:
Fixed spacing

Replaced tabs with spaces

adorostkar committed:
Spacing correction

More spacing correction

tgamblin committed:
Fix some bugs in architecture.

eschnett committed:
Update HDF5 to version 1.10.0-patch1

nrichart committed:
Adding package fenics

becker33 committed:
Fixed config format to allow default compiler flags

nrichart committed:
Adding a version to tetgen

nrichart committed:
Corrections for flake8

nrichart committed:
Introduced a bug by correcting flake8 warnings

nrichart committed:
Addapting the mvapich2@2.1: configure options for slurm

nrichart committed:
Correcting the flake8 warnings

tgamblin committed:
Merge pull request #1011 from epfl-scitas/packages/fenics

Adding package FEniCS

tgamblin committed:
Merge pull request #1010 from epfl-scitas/packages/tetgen

Adding a version to tetgen

tgamblin committed:
Merge pull request #1008 from LLNL/features/cflags-fix

Fixed config format to allow default compiler flags

tgamblin committed:
Merge pull request #1005 from eschnett/eschnett/hdf5-update

Update HDF5 to version 1.10.0-patch1

tgamblin committed:
Merge pull request #1001 from ashkan2200/develop

Dealii, fixed issue with tests

tgamblin committed:
Merge pull request #1014 from epfl-scitas/packages/mvapich2

Changes to take into account slurm in version @2.1: of mvapich2

tgamblin committed:
Update version to correspond to master.

mamelara committed:
Changed how architecture looks like in spec.yaml. Now it's only three strings, platform, os, target in spec.yaml. Also did some flake8 formatting changes

mamelara committed:
Merge branch 'features/newarch' of https://github.com/NERSC/spack into features/newarch

mamelara committed:
Made a merge commit earlier with Todd fixing all_platforms, but changed how architecture looks in spec.yaml. Just three strings. Also made some flake8 formatting changes

trws committed:
Merge pull request #869 from brettviren/feature/views

Feature/views

mamelara committed:
Passing architecture to functions to find compiler. We can use the platform name if provided to find the operating system. Also changed error to operating system instead of target since operating system is now in charge of compiler finding.

mamelara committed:
Passing full arch for proper compiler spec so that we can use the platform to get the operating system

mamelara committed:
Passing through full arch instead of just operating sysystem so that we can use the platform to get the operating system

mamelara committed:
Got rid of unnecessary global keyword

mamelara committed:
Changed OS name to CNL10 so that we properly find the compilers for the test platform

mamelara committed:
Changed to passing full architecture spec to function rather than just platform_os so we can grab the operating system from the platform getters

mamelara committed:
Load external modules

mamelara committed:
Added to_dict back into operating system

lee218llnl committed:
put older versions back + version-specific dwarfdump dir

lee218llnl committed:
flake8 fixes

alalazo committed:
lmod : updated package and dependencies

alalazo committed:
lmod : fixed issues with LUA_PATH during installation

alalazo committed:
qa : fixed flake8 issues

Commit:
Fixes issue #438

Commit:
Fixed flake8 formatting problems

opoplawski committed:
Add patch to compile pcre with Intel 16.0.3 compiler

opoplawski committed:
Add additional blank line for flake8

KineticTheory committed:
Package serf also depends on zlib.

+ Add zlib as a required dependency.
+ Point scons build system to the spack provided zlib installation.

KineticTheory committed:
Fix style issues.

tgamblin committed:
Merge pull request #1020 from LLNL/bugfix/libdwarf_versions

put older versions back + version-specific dwarfdump dir

becker33 committed:
changed compiler yaml format

becker33 committed:
I made a hash of that minor merge
Merge commit '4eeb' into features/newarch

alalazo committed:
versions : added unit tests for the new properties

alalazo committed:
qa : fixed style and a bug reported by flake8

lib/spack/spack/test/versions.py:313: [F811] redefinition of unused 'test_union_with_containment' from line 306
Flake8 found errors.

eschnett committed:
Do not pass "/usr" or "/usr/local" as path to configure

trws committed:
Merge pull request #1026 from epfl-scitas/packages/lmod

lmod : updated package and dependencies

KineticTheory committed:
Make Boost an optional feature of Trilinos.

I have been unable to build Boost with some toolset/platform combinations. The
users I support don't need the Boost features in Trilinos, so I have been
manually hacking trilinos/package.py to disable boost. I would like to propose
that these changes be added to the released version.

+ Add boost as a variant, default to true so there is no change to existing
  behavior.
+ Move the CMake configure options related to Boost into a conditional block
  that is triggered by "if '+boost' in spec".

KineticTheory committed:
Update formatting to meet code style requirements.

daboehme committed:
Use https for github

eschnett committed:
Add packages c-blosc and hdf5-blosc

c-blosc provides the Blosc compression filter, and hdf5-blosc provides a plugin for HDF5 to use that filter.

eschnett committed:
Correct Blosc install on Darwin

eschnett committed:
Install shared libraries correctly on Linux

tgamblin committed:
Merge pull request #1051 from daboehme/develop

Use https for github in Caliper package

tgamblin committed:
Merge pull request #1031 from opoplawski/pcre

Add patch to compile pcre with Intel 16.0.3 compiler

tgamblin committed:
Merge pull request #1037 from KineticTheory/develop

Package serf also depends on zlib.

tgamblin committed:
Merge pull request #1040 from KineticTheory/pkg-trilinos

Make Boost an optional feature of Trilinos.

tgamblin committed:
Merge pull request #1028 from JRWynneIII/boost-graph-parallel-fix

Add variant for boost graph and parallel graph

tgamblin committed:
Merge pull request #1058 from eschnett/eschnett/blosc

Add packages c-blosc and hdf5-blosc

tgamblin committed:
Merge pull request #1045 from eschnett/eschnett/openmpi-verbs-usr

Do not pass "/usr" or "/usr/local" as path to configure

becker33 committed:
Two minor fixes

becker33 committed:
two minor fixes

tgamblin committed:
Add `__contains__` to arch to satsify old packages.

tgamblin committed:
be more tolerant when parsing new specs.

tgamblin committed:
Add libxcb dependencies.  Fix up packages to get py-basemap and stat to build.

tgamblin committed:
fix up packages for new architecture.

tgamblin committed:
Just call old os "unknown"

tgamblin committed:
Fixes #460: Do not show variants by default in spack find.

This does two things:

1. By default `spack find` no longer shows variants.  You have to
   supply `-v` to get that

2. This improves the `colify` implementation so that it no longer pads
   the rightmost column.  This avoids the issue where if one spec was
   too long in the output, *all* specs would have space padding added
   to that width, and it would look like the output of `spack find`
   was double spaced.  This no longer happens -- the one bad line
   wraps around and the other lines are now single-spaced when you use
   `-v` with boost.

tgamblin committed:
Merge branch 'features/newarch' into develop

Conflicts:
	lib/spack/spack/config.py
	var/spack/repos/builtin/packages/lua/package.py

mamelara committed:
Added str method to mac_osx so that whenever we try to access darwins different operating systems we can grab the name correctly. This avoids having to worry whether we have a version attached to it or not.

tgamblin committed:
rename Mac OS X to macos

tgamblin committed:
simplify instance check.

tgamblin committed:
remove dead code, cleanup.

tgamblin committed:
Replace 2.6-incompatible skipIf with a conditional.

- skipIf doesn't work with Python 2.6.

gartung committed:
Merge branch 'llnl-develop' into binary_packages

Conflicts:
	lib/spack/spack/cmd/install.py
	lib/spack/spack/package.py
	lib/spack/spack/test/__init__.py

gartung committed:
fix typo from merge

gartung committed:
fixup merge
